src/vanilla/Model/CodeModelTS.cs

```diff
@@ -588,7 +591,8 @@ public virtual void ConstructRuntimeImportForModelIndex(TSBuilder builder)
 
         public virtual void PackageDependencies(JSONObject dependencies)
         {
-            dependencies.StringProperty("@azure/ms-rest-js", "^2.0.4");
+            dependencies.StringProperty("@azure/ms-rest-js", "^2.2.0");
+            dependencies.StringProperty("@azure/core-auth", "^1.1.4");
```

Comment:
I think this dependency should not be added unless there is a credential, right? Is there a simple way to check for this condition here? cc @joheredi

Reply:
I think you could check either Settings.AddCredentials or Settings.AzureArm